core/src/main/kotlin/org/jetbrains/research/testspark/core/generation/llm/prompt/PromptBuilder.kt

```diff
@@ -118,6 +118,10 @@ internal class PromptBuilder(private var prompt: String) {
                     fullText += "${currentSubClass.qualifiedName} is $subClassTypeName ${entry.key.qualifiedName}.\n"
                 }
             }
+            // If we have some information added for polymorphism relations, we add an instruction
+            if (fullText.isNotBlank()) {
+                fullText = "Use the following information about polymorphic relationship of classes that exist in the project. Don't use mocking for these classes:\n$fullText" // TODO
```

Comment:
I cannot understand what exactly is instructed in this prompt. Could you explain? **(1)**

If I understood correctly the problem mentioned in the [issue](https://github.com/JetBrains-Research/TestSpark/issues/254), LLM thinks that the entry `class X is a sub-class of Y` means that there is no class `X`, and we want the LLM to create this `X` and make it a sub-class of `Y`. Is it correct? **(2)**

There are several questions:
  1. Is this prompt required to emphasize that the classes `X` are **present in the project** and the LLM **must not create mocks** for classes `Y`? **(3)**
2. `Don't use mocking for these classes` - for what classes: classes `X` or classes `Y`? **(4)**

How about something like this:
_"Consider the following polymorphic relationships of classes present in the project. Use them for instantiation when necessary. Do not mock classes if an instantiation of a sub-class is applicable"_ 
OR:
_"Use the following polymorphic class relationships [present in the project] for instantiation where applicable. Avoid mocking when a subclass instantiation is possible"_

You may use **(i)** to refer to questions.

Reply:
(1) Your following suggestions for instruction are more reflective, so I have updated them.
(2) and (3) Yes. Without instruction, it just mocks the class and ignores the sub-class. With not mentioning it is available in the project, it often generates both the sub-class and the tests.
(4) for sub class. In general, your suggestions are better. I updated the instructions according to them. Thanks :)